src/Model/Resolver/Products/DataPostProcessor/Attributes.php

```diff
@@ -253,38 +254,43 @@ protected function appendWithOptions(
                 $values = array_flip( // Flip Array
                     array_merge( // phpcs:ignore
                         array_filter( // explode might return array with empty value, remove such values
-                            explode(',', $productAttributes[$id][$key]['attribute_value'] ?? '')
+                            explode(',', $productAttributes[$productId][$attributeCode]['attribute_value'] ?? '')
                         ),
                         $variantAttributeValues
                     )
                 );
 
-                $options = $attribute->getOptions();
-                array_shift($options);
-                $productAttributes[$id][$key]['attribute_options'] = [];
+                $productAttributes[$productId][$attributeCode]['attribute_options'] = [];
```

Comment:
I'm not sure why the original code had array_shift on the $options; your new version doesn't have an equivalent
any ideas on what it was for?

Reply:
Tho, it appears that first element in array from getOption is empty (only if attribute has options).